chainio/clients/avsregistry/reader.go

```diff
@@ -88,29 +88,36 @@ type AvsRegistryReader interface {
 	) (map[types.OperatorId]types.Socket, error)
 }
 
-type AvsRegistryChainReader struct {
+type Config struct {
+	RegistryCoordinatorAddress    common.Address
+	OperatorStateRetrieverAddress common.Address
+}
+
+type ChainReader struct {
 	logger                  logging.Logger
-	blsApkRegistryAddr      gethcommon.Address
-	registryCoordinatorAddr gethcommon.Address
+	blsApkRegistryAddr      common.Address
+	registryCoordinatorAddr common.Address
 	registryCoordinator     *regcoord.ContractRegistryCoordinator
 	operatorStateRetriever  *opstateretriever.ContractOperatorStateRetriever
 	stakeRegistry           *stakeregistry.ContractStakeRegistry
 	ethClient               eth.Client
 }
 
 // forces AvsReader to implement the clients.ReaderInterface interface
-var _ AvsRegistryReader = (*AvsRegistryChainReader)(nil)
+var _ Reader = (*ChainReader)(nil)
 
-func NewAvsRegistryChainReader(
-	registryCoordinatorAddr gethcommon.Address,
-	blsApkRegistryAddr gethcommon.Address,
+func NewChainReader(
+	registryCoordinatorAddr common.Address,
+	blsApkRegistryAddr common.Address,
 	registryCoordinator *regcoord.ContractRegistryCoordinator,
 	operatorStateRetriever *opstateretriever.ContractOperatorStateRetriever,
 	stakeRegistry *stakeregistry.ContractStakeRegistry,
 	logger logging.Logger,
 	ethClient eth.Client,
-) *AvsRegistryChainReader {
-	return &AvsRegistryChainReader{
+) *ChainReader {
+	logger = logger.With("module", "avsregistry/ChainReader")
```

Comment:
let's create a `logger.sublogger("avsregistry/ChainReader")` function which automatically adds the "module" tag? Otherwise we might end up with people using different keys in different subloggers

Reply:
good call.